xtra-libp2p/src/dialer.rs

```diff
@@ -35,22 +36,24 @@ impl Actor {
         }
     }
 
-    /// Returns error if we cannot access Endpoint or the connect address is not
-    /// present inside Endpoint.
+    /// Returns error if we cannot access Endpoint or the peer is not connected Endpoint.
     async fn check_connection_active_in_endpoint(&self) -> Result<(), Error> {
         let ConnectionStats {
-            listen_addresses, ..
+            connected_peers, ..
         } = self
             .endpoint
             .send(GetConnectionStats)
             .await
             .map_err(|_| Error::NoEndpoint)?;
 
-        dbg!(&listen_addresses);
-        dbg!(&self.connect_address);
-
-        listen_addresses
-            .contains(&self.connect_address)
+        connected_peers
+            .contains(
+                &self
+                    .connect_address
+                    .clone()
+                    .extract_peer_id()
+                    .ok_or(Error::InvalidPeerId)?,
```

Comment:
🔧 It's annoying to have to deal with this error when the result will never change from the moment we construct the actor. Can we get around this?

Reply:
We could build and save the `PeerId` in `new` and make `new` fallible. Only downside is slight redundancy of data. But I prefer that over having to handle errors every time we check if a connection is active.

Reply:
unfortunately, the supervisor does not support fallible actor constructors. I had a quick stab in remodelling the arguments to tighten the API, but it's a bit annoying too, and I don't want to use `expect()` anywhere. Will keep it as-is for now, might get back to it later on.